internal/apps/ai-proxy/models/prompt/dbclient.go

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prompt
+
+import (
+	"context"
+
+	"gorm.io/gorm"
+
+	"github.com/erda-project/erda-proto-go/apps/aiproxy/prompt/pb"
+	commonpb "github.com/erda-project/erda-proto-go/common/pb"
+	"github.com/erda-project/erda/internal/apps/ai-proxy/models/client_model_relation"
+	"github.com/erda-project/erda/internal/apps/ai-proxy/models/common"
+	"github.com/erda-project/erda/internal/apps/ai-proxy/models/message"
+	"github.com/erda-project/erda/internal/apps/ai-proxy/models/metadata"
+)
+
+type DBClient struct {
+	DB *gorm.DB
+}
+
+func (dbClient *DBClient) Create(ctx context.Context, req *pb.PromptCreateRequest) (*pb.Prompt, error) {
+	tx := dbClient.DB.Begin()
+	// check client id
+	if req.ClientId != "" {
+		if err := client_model_relation.TxCheckClientID(tx, req.ClientId); err != nil {
+			tx.Rollback()
+			return nil, err
+		}
+	}
+	// create
+	c := &Prompt{
+		Name:     req.Name,
+		Desc:     req.Desc,
+		ClientID: req.ClientId,
+		Messages: message.FromProtobuf(req.Messages),
+		Metadata: metadata.FromProtobuf(req.Metadata),
+	}
+	if err := tx.Model(c).Create(c).Error; err != nil {
+		tx.Rollback()
+		return nil, err
+	}
+	// commit
+	tx.Commit()
```

Comment:
Is tx.Commit() need to be call before exit when some error happens?

Reply:
No, `rollback` invoked if normal error occurred.